server/server.go

```diff
@@ -139,22 +141,30 @@ func (g *ReleaseManager) CheckForUpdate(p *Params) (res *Result, err error) {
 		return nil, ErrNoUpdateAvailable
 	}
 
-	// Looking for the asset thay matches the current app checksum.
-	var current *Asset
-	if current, err = g.lookupAssetWithChecksum(p.OS, p.Arch, p.Checksum); err != nil {
-		// No such asset with the given checksum, nothing to compare. Making the
-		// client download the whole binary.
+	// A newer version is available!
 
-		r := &Result{
+	fullResult := func(update *Asset) *Result {
+		return &Result{
 			Initiative: INITIATIVE_AUTO,
 			URL:        update.URL,
 			PatchType:  PATCHTYPE_NONE,
 			Version:    update.v.String(),
 			Checksum:   update.Checksum,
 			Signature:  update.Signature,
 		}
+	}
+
+	if p.OS == OS.Android {
+		// if Android, stop here and return the full binary to the client
+		return fullResult(update), nil
+	}
 
-		return r, nil
+	// Looking for the asset thay matches the current app checksum.
+	var current *Asset
+	if current, err = g.lookupAssetWithChecksum(p.OS, p.Arch, p.Checksum); err != nil {
```

Comment:
Can we instead modify `lookupAssetWithChecksum` so it returns an error when `p.OS` is `OS.Android`?